ci/jenkinsfiles/common-lib.groovy

```diff
@@ -0,0 +1,241 @@
+  if (tag) {
+    container('maven') {
+      version = sh(returnStdout: true, script: "npx --ignore-existing semver -i prerelease --preid ${preid} ${tag} | tr -d '\n'")
+    }
+  }
+  return version
+}
+
+String getPullRequestVersion() {
+  return "${readMavenPom().getVersion()}-${BRANCH_NAME}"
+}
+
+void gitCheckout(String reference) {
+  sh "git checkout ${reference}"
+}
+
+void gitCommit(String message, String option) {
+  echo '''
+    ----------------------------------------
+    Git Commit
+    ----------------------------------------
+  '''
+  sh """
+    git commit ${option} -m "${message}"
+  """
+}
+
+void gitPush(String reference) {
+  sh "git push origin ${reference}"
+}
+
+void gitTag(String tagname, String message) {
+  sh """
+    git tag -a ${tagname} -m "${message}"
+  """
+}
+
+def isPullRequest() {
+  return "${BRANCH_NAME}" =~ /PR-.*/
+}
+
+void lint() {
+  echo '''
+    ----------------------------------------
+    Run Linting Validations
+    ----------------------------------------
+  '''
+  sh """
+    cd ${FRONTEND_FOLDER}
+    npm run lint
+  """
+}
+
+String mavenArgs() {
+  String args = '-DskipITs=true -fae -B -nsu'
+  return args
+}
+
+boolean needsSaucelabs() {
+  return this.isPullRequest() && pullRequest.labels.contains('saucelabs')
+}
+
+void runFrontEndUnitTests() {
+  return {
+    stage('Run Unit tests: FrontEnd') {
+      container('maven') {
+        String context = 'coldstorage/utests/frontend'
+        String message = 'Unit tests - FrontEnd'
+        echo '''
+          ----------------------------------------
+          Run FrontEnd Unit tests
+          ----------------------------------------
+        '''
+        script {
+          this.setGitHubBuildStatus("${context}", "${message}", 'PENDING', "${GIT_URL}")
+          String sauceAccessKey = ''
+          String sauceUsername = ''
+          if (this.needsSaucelabs()) {
+            String saucelabesSecretName = 'saucelabs-coldstorage'
+            sauceAccessKey =
+              sh(script: "jx step credential -s ${saucelabesSecretName} -k key", returnStdout: true).trim()
+            sauceUsername =
+              sh(script: "jx step credential -s ${saucelabesSecretName} -k username", returnStdout: true).trim()
+          }
+          try {
+            withEnv(["SAUCE_USERNAME=${sauceUsername}", "SAUCE_ACCESS_KEY=${sauceAccessKey}"]) {
+              container('playwright') {
+                sh """
+                  cd ${FRONTEND_FOLDER}
+                  npm clean-install
+                  npm run test
+                """
+              }
+            }
+            this.setGitHubBuildStatus("${context}", "${message}", 'SUCCESS', "${GIT_URL}")
+          } catch (err) {
+            this.setGitHubBuildStatus("${context}", "${message}", 'FAILURE', "${GIT_URL}")
+            throw err
+          }
+        }
+      }
+    }
+  }
+}
+
+void runBackEndUnitTests() {
+  return {
+    stage('Run Unit tests: BackEnd') {
```

Comment:
same thing on the returned type `void` vs returning an object.  same than [Here](https://github.com/nuxeo/nuxeo-coldstorage/pull/3/files#r526242648)

Reply:
note I also noted the same thing on Retention, we will try to clean it later on one of our tickets like crowding